src/data_structures/graph.ml

```diff
@@ -0,0 +1,44 @@
+type node =
+  { ind : int
+  ; info : string option
+  ; children : int list
+  ; mutable visited : bool
+  }
+
+type t =
+  { nodes : node list
+  ; entry_points : int list
+  }
+
+let empty = { nodes = []; entry_points = [] }
+
+let set_entry_points graph e = { nodes = graph.nodes; entry_points = e }
```

Comment:
```ocaml
let set_entry_points graph entry_points = { graph with entry_points }
```

Also, I believe we should add a node for each entry point even if it has no edge, right?